tests/integration/server/api/users/index.get.test.ts

```diff
@@ -0,0 +1,53 @@
+import { describe, expect, it, vi, afterEach } from "vitest"
+import { setup, $fetch } from "@nuxt/test-utils/e2e"
+
+describe("Api Route /api/users", async () => {
+  await setup()
+  afterEach(() => {
+    vi.resetAllMocks()
+  })
+
+  it("Should return a list of users", async () => {
+    const response = await $fetch("/api/users")
+    // Expected output structure
+    const expectedStructure = [
+      { username: "Admin", role: "ADMIN", verified: true },
+      { username: "Moderator", role: "MODERATOR", verified: true },
+      { username: "User", role: "USER", verified: true },
+    ]
+
+    // Ensure data has length of at least 3 (migration with seeds work)
+    expect(response.length).toBeGreaterThan(3)
+
```

Comment:
Add an `expectTypeOf(response).toEqualTypeOf<UserDetail>()` or similar to check the correct type and instead remove a part of the responses.forEach section

Reply:
cant get this to work:

Type 'UserList' does not satisfy the constraint '{ [x: number]: { username: "Expected: never, Actual: string"; id: "Expected: never, Actual: string"; role: "Expected: never, Actual: string"; verified: "Expected: never, Actual: boolean"; }; 
username: "Expected: string, Actual: never"; id: "Expected: string, Actual: never"; role: "Expected: literal string: USER, Act...'.
  Types of property 'username' are incompatible.
    Type 'string' is not assignable to type '"Expected: string, Actual: never"'.

Reply:
Hmm yeah, it should be a list of list objects not one detail object here, so `expectTypeOf(response).toEqualTypeOf<UserList[]>()` and then the user role in server/db/models/user.ts needs to be cast to a UserRole for the types to match, (`user.role as UserRole`). I didn't test the last one, maybe it needs some more sophisticated conversion.